source/fundamentals/data-formats/document-data-format-extended-json.txt

```diff
@@ -119,146 +119,280 @@ corresponds to the format of the example you want to see:
+      :java-docs:`JsonReader </apidocs/bson/org/bson/json/JsonReader.html>` class.
+      This class contains methods to sequentially parse the fields and values
+      in any format of the Extended JSON string, and returns them as Java objects.
+      MongoDB driver's document classes also use this class to parse Extended JSON.
+      
+      The following code example shows how you can use ``JsonReader`` to convert
+      an Extended JSON string into Java objects:
+      
+      .. code-block:: java
+      
+         String ejsonStr = "{ \"_id\": { \"$oid\": \"507f1f77bcf86cd799439011\"}," +
+                           "\"myNumber\": {\"$numberLong\": \"4794261\" }}}";
+      
+         JsonReader jsonReader = new JsonReader(ejsonStr);
+      
+         jsonReader.readStartDocument();
+      
+         jsonReader.readName("_id");
+         ObjectId id = jsonReader.readObjectId();
+         jsonReader.readName("myNumber");
+         Long myNumber = jsonReader.readInt64();
+      
+         jsonReader.readEndDocument();
+      
+         System.out.println(id + " is type: " + id.getClass().getName());
+         System.out.println(myNumber + " is type: " + myNumber.getClass().getName());
+      
+         jsonReader.close();
+      
+      The output of this code example resembles the following:
+      
+      .. code-block:: none
+         :copyable: False
+      
+         507f1f77bcf86cd799439011 is type: org.bson.types.ObjectId
+         4794261 is type: java.lang.Long
+      
+      
+      For additional information on the classes and methods mentioned on this
+      section, see the following API documentation:
+      
+   .. tab:: JsonObject Class
+      :tabid: jsonobject-class-read
+
+      You can read an Extended JSON string into a Java ``JsonObject`` object through the ``JsonObject``
+      class constructor. The :java-docs:`JsonObject <apidocs/bson/org/bson/json/JsonObject.html>`
+      class acts as a wrapper for a JSON string. 
+      
+      The following example shows instantiating a ``JsonObject`` with an Extended JSON string:
+
+      .. code-block:: java
+      
+         String ejsonStr = "{ \"_id\": { \"$oid\": \"507f1f77bcf86cd799439011\"}," +
+                           "\"myNumber\": {\"$numberLong\": \"4794261\" }}}";
+         JsonObject jsonObj = new JsonObject(ejsonStr);
+         System.out.println(jsonObj.getJson());
+
+      The output of this code example resembles the following:
+      
+      .. code-block:: none
+         :copyable: False
+      
+         { "_id": { "$oid": "507f1f77bcf86cd799439011"},"myNumber": {"$numberLong": "4794261" }}}
+
+      If you want to work primarily with raw JSON strings and avoid intermediate ``Document`` maps in your deployment,
+      the ``JsonObject`` class is a more performant alternative than reading JSON to a ``Document`` instance and
+      calling the ``toJson()`` method on that ``Document`` instance. ``JsonObject`` is discussed in more depth
```

Comment:
I feel this could be cleaned up a bit. You could also be more direct and prescriptive.

>If you want to work primarily with raw JSON strings and avoid intermediate ``Document`` maps in your deployment, use the ``JsonObject`` class.

Reply:
Thank you for the note @cbush. I agree that your suggested alternative is cleaner. In my first draft, I think I wanted to try and explain to the reader why the `JsonObject` is a more **performant** alternative, however on rereading with your notes I think it suffices to be prescriptive and just tell the reader to use `JsonObject` in a certain scenario and not go into too much depth as to why that is the case.

I'll try to develop a sense of when and when not to go into a technical explanation when describing a feature/workflow.